include/mega/node.h

```diff
@@ -110,7 +110,7 @@ struct Fingerprints
 // filesystem node
 struct MEGA_API Node : public NodeCore, FileFingerprint
 {
-    MegaClient* client;
+    MegaClient* client = nullptr;
```

Comment:
I don't think it can ever be null in practice though?   setting it null here is a bit misleading?

Reply:
This initialization is not used when the constructor initializer list is used. If not, then this is the default as opposed to some random memory address. I think this makes code well defined and predictable. 

Also, I have unit tests coming up in another branch that require the client to be null. We need this so that we can circumvent certain code that requires a valid client. Note that the client can currently not be mocked out. Alternatively, we could introduce a ClientInterface but that's a lot of changes we might not want.

Reply:
I like class members (specially pointers) to be always initialized. IMHO, I don't mind if it's set in there, but it's also true that the only user-defined `Node` ctor accepts and sets the `client`. If you have a branch where the `Node` is not created with:
```Node::Node(MegaClient* cclient, node_vector* dp, handle h, handle ph,
           nodetype_t t, m_off_t s, handle u, const char* fa, m_time_t ts)```
then this initialization makes sense in that branch, not here.